tests/main.test.ts

```diff
@@ -47,22 +47,66 @@ describe("src/main.ts", function() {
 
     const validAny = validModel(anyType);
 
-    it('should allow all types when no restriction', function(done) {
+    it('should allow all types for -any- schema when there are no restriction',
+    function(done) {
       const validTypes = [
         undefined,
         null,
         5,
         "string",
         true,
-        {first: 1, second: "two"},
-        // null,
-        // undefined
+        {first: 1, second: "two"}
       ];
 
-      mochaAssertPromiseResovles(
-        Promise.all(validTypes.map(validAny)),
+      mochaAssertPromiseResovles(Promise.all(validTypes.map(validAny)), done);
+    });
+
+    const anyWithBoolRestriction: anySchema = {
```

Comment:
"do you want the restriction to also work with any type?" 

It already does, that's what I'm referring to. It doesn't force anything to be a boolean 
right now. When you check `if(thing)` in TS that doesn't mean the `thing` has to be a   
boolean. If you want to specifcally just have this be for booleans, you'd need to also check `typeof x === "boolean"`.

Right now you're restriction is just checking that whatever you pass is *not* a "truthy" value, and so you should name it accordingly. `anyNonTruthyValueSchema` would be more appropriate for instance.